### PR TITLE
Post tutorial

### PR DESCRIPTION
**Pull request by dprats**
This PR adds two more code samples to the repo:

1. Demo canister in Motoko that sends an HTTP `POST` request
2. Demo canister in Rust that sends an HTTP `POST` request

This PR is related to this one: https://github.com/dfinity/portal/pull/1569

More specifically:

## Motoko
* creates `motoko/send_http_post` sample dapp for sending `POST` in motoko
* creates`motoko/send_http_get` sample dapp for sending `GET` in motoko
* deletes old `motoko/https_outcalls_GET_request` sample dapp
* deletes old `motoko/https_outcalls_POST_request` sample dapp
* adds CI tests for `motoko/send_http_post`
* adds CI tests for `motoko/send_http_get`

## Rust
* creates `rust/send_http_post` sample dapp for sending `POST` in rust
* creates `rust/send_http_get` sample dapp for sending `GET` in rust
* deletes old `rust/https_outcalls_GET_request` sample dapp
* adds CI tests for `rust/send_http_post`
* adds CI tests for `rust/send_http_get`

### CODE REVIEW

File: .github/workflows/motoko-send-http-get-skip.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: motoko-send-http-get
+on:
+  pull_request:
+    paths-ignore:
+      - motoko/send-http-get/**
```
Comment (ericswanson-dfinity) [inline .github/workflows/motoko-send-http-get-skip.yml:5]: Needs to match the actual path
```suggestion
      - motoko/send_http_get/**
```

File: .github/workflows/motoko-send-http-post-skip.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: motoko-send-http-post
+on:
+  pull_request:
+    paths-ignore:
+      - motoko/send-http-post/**
```
Comment (ericswanson-dfinity) [inline .github/workflows/motoko-send-http-post-skip.yml:5]:
```suggestion
      - motoko/send_http_post/**
```

File: motoko/https_outcalls_GET_request/README.md
Comment (krpeacock) [inline motoko/https_outcalls_GET_request/README.md:1]: Was this and the `https_outcalls_GET_request` declarations moved somewhere else, or are they just no longer relevant?

File: .github/workflows/rust-send-http-get-skip.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: rust-send-http-get
+on:
+  pull_request:
+    paths-ignore:
+      - rust/send-http-get/**
```
Comment (ericswanson-dfinity) [inline .github/workflows/rust-send-http-get-skip.yml:5]:
```suggestion
      - rust/send_http_get/**
```

File: .github/workflows/rust-send-http-post-skip.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+name: rust-send-http-post
+on:
+  pull_request:
+    paths-ignore:
+      - rust/send-http-post/**
```
Comment (ericswanson-dfinity) [inline .github/workflows/rust-send-http-post-skip.yml:5]:
```suggestion
      - rust/send_http_post/**
```

File: motoko/https_outcalls_GET_request/README.md
Comment (dprats) [reply motoko/https_outcalls_GET_request/README.md:1]: no longer relevant

File: motoko/send_http_get/Makefile [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+.PHONY: upgrade
+.SILENT: upgrade
+upgrade: build
+	dfx canister install send_http_get_backend --mode=upgrade
+
+.PHONY: test
+.SILENT: test
+test: install
+	dfx canister call send_http_get_backend get_icp_usd_exchange \
+		| grep '[[1682978460,5.714,5.718,5.714,5.714,243.5678]]' && echo 'PASS'
```
Comment (ericswanson-dfinity) [inline motoko/send_http_get/Makefile:30]: In a regular expression:
- `[ab.c]` means match any of `a`, `b`, or `c`, or '.,'
- `.` outside of square brackets means match any character
So as written, this grep expression means something like:
`string contains any of 0 1 2 3 4 5 6 7 8 9 . followed by ]`
```
$ echo "5]" \                                             
                | grep '[[1682978460,5.714,5.718,5.714,5.714,243.5678]]' && echo 'PASS'
5]
PASS
$ echo ".]" \
                | grep '[[1682978460,5.714,5.718,5.714,5.714,243.5678]]' && echo 'PASS'
.]
PASS

```
If you are looking for single brackets around these values:
```suggestion
		| grep '\[1682978460,5\.714,5\.718,5\.714,5\.714,243\.5678\]' && echo 'PASS'
```

If you are looking for double brackets around these values:
```suggestion
		| grep '\[\[1682978460,5\.714,5\.718,5\.714,5\.714,243\.5678\]\]' && echo 'PASS'
```



```
$ echo "[[1682978460,5.714,5x718,5.714,5.714,243.5678]]" \
                | grep '\[1682978460,5\.714,5\.718,5\.714,5\.714,243\.5678\]' && echo 'PASS'
$ echo "[[1682978460,5.714,5.718,5.714,5.714,243.5678]]" \ 
                | grep '\[1682978460,5\.714,5\.718,5\.714,5\.714,243\.5678\]' && echo 'PASS'
[[1682978460,5.714,5.718,5.714,5.714,243.5678]]
PASS
```

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

File: rust/send_http_get/Makefile [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+.PHONY: upgrade
+.SILENT: upgrade
+upgrade: build
+	dfx canister install send_http_get_backend --mode=upgrade
+
+.PHONY: test
+.SILENT: test
+test: install
+	dfx canister call send_http_get_backend get_icp_usd_exchange \
+		| grep '[[1682978460,5.714,5.718,5.714,5.714,243.5678]]' && echo 'PASS'
```
Comment (ericswanson-dfinity) [inline rust/send_http_get/Makefile:30]: Same issue as motoko get test

File: rust/send_http_get_rust/.gitignore [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Various IDEs and Editors
```
Comment (ericswanson-dfinity) [inline rust/send_http_get_rust/.gitignore:1]: why do we have `rust/send_http_get/` and `rust/send_http_get_rust/` ?

File: rust/send_http_post/Makefile [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+.SILENT: upgrade
+upgrade: build
+	dfx canister install send_http_post_backend --mode=upgrade
+
+.PHONY: test
+.SILENT: test
+test: install
+	# echo 'PASS dummy test'
+	dfx canister call send_http_post_backend send_http_post_request \
+		| grep 'success' && echo 'PASS'
```
Comment (ericswanson-dfinity) [inline rust/send_http_post/Makefile:32]: at minimum
```suggestion
		| grep 'success.*true' && echo 'PASS'
```

File: rust/send_http_post_rust/.gitignore [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Various IDEs and Editors
```
Comment (ericswanson-dfinity) [inline rust/send_http_post_rust/.gitignore:1]: Same question about `rust/send_http_post/` vs `rust/send_http_post_rust/`

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

File: motoko/send_http_post_motoko/README.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
+# send_http_post_motoko
```
[the remark this answers is not in the document]
Comment (dprats) [reply motoko/send_http_post_motoko/README.md:1]: we should not, was that file not deleted? let me fix

File: rust/send_http_post_rust/.gitignore (hunk 9 above)
Comment (dprats) [reply rust/send_http_post_rust/.gitignore]: Good call. They were removed on local branch, but changes were not pushed up. Pushed up now.

File: motoko/send_http_post_motoko/README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (dprats) [reply motoko/send_http_post_motoko/README.md]: good call. this is fixed now.

File: rust/send_http_get/Makefile (hunk 6 above)
Comment (dprats) [reply rust/send_http_get/Makefile:30]: added

File: motoko/hello/Makefile [hunk 11]
```diff
@@ -22,12 +22,6 @@ install: build
 upgrade: build
 	dfx canister install --all --mode=upgrade
 
-.PHONY: test
-.SILENT: test
-test: install
-	dfx canister call hello greet '("everyone")' \
-		| grep '("Hello, everyone!")' && echo 'PASS'
-
```
Comment (ericswanson-dfinity) [inline motoko/hello/Makefile:30]: Was this intentionally removed?

Comment (dprats) [reply motoko/hello/Makefile:30]: no it was an accident, need to add back

**Review by krpeacock — APPROVED** [review]

File: motoko/hello/Makefile [hunk 12]
```diff
@@ -20,13 +20,7 @@ install: build
 .PHONY: upgrade
 .SILENT: upgrade
 upgrade: build
-	dfx canister install --all --mode=upgrade
-
-.PHONY: test
-.SILENT: test
-test: install
-	dfx canister call hello greet '("everyone")' \
-		| grep '("Hello, everyone!")' && echo 'PASS'
```
Comment (ericswanson-dfinity) [inline motoko/hello/Makefile:29]: this test is still missing

Comment (dprats) [reply motoko/hello/Makefile]: ah good catch! thanks fixing

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**